core/src/test/scala/unit/kafka/zk/EmbeddedZookeeper.scala

```diff
@@ -39,6 +39,16 @@ class EmbeddedZookeeper() {
   def shutdown() {
     CoreUtils.swallow(zookeeper.shutdown())
     CoreUtils.swallow(factory.shutdown())
+
+    def isDown(): Boolean = {
```

Comment:
`ZooKeeperTestHarness.tearDown` calls this method:

```scala
if (zookeeper != null)
        CoreUtils.swallow(zookeeper.shutdown())
```

Isn't that good enough?